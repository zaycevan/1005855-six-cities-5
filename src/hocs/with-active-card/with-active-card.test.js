import renderer from "react-test-renderer";
import withActiveCard from "./with-active-card";

const noop = () => {};

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withActiveCard(MockComponent);

const tree = renderer.create(
    <MockComponentWrapped
      activeCardId={1}
      onActiveCard={noop}
      outActiveCard={noop}
    >
      <></>
    </MockComponentWrapped>
);

it(`Should withActiveCard render correctly`, () => {
  expect(tree.toJSON()).toMatchSnapshot();
});
