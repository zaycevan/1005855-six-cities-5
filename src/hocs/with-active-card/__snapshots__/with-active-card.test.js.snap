// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should withActiveCard render correctly 1`] = `<div />`;
